python/cudf/cudf/utils/ioutils.py

```diff
@@ -830,16 +841,24 @@ def get_filepath_or_buffer(path_or_data, compression, iotypes=(BytesIO)):
     compression : str
         Type of compression algorithm for the content
     """
-    if is_url(path_or_data):
-        with urllib.request.urlopen(path_or_data) as url:
-            compression = url.headers.get("Content-Encoding", None)
```

Comment:
New logic does not infer compression from urls based on the Content-Encoding header. Need to check `fsspec` to see if this is possible

Reply:
requests does this automatically for the standard compressions (gzip and deflate only, I think)